### DISCUSSION TITLE
[plugin:vite:vue] <template v-for> key should be placed on the <template> tag.

### DISCUSSION THREAD

**Discussion by afurosan** [quasarframework/quasar#16643] [General - Components / Directives / etc] [answered]
Hi!
In the following program, The following program produces an error "[plugin:vite:vue] <template v-for> key should be placed on the <template> tag."
However, the ":key="i" is specified within the template.

Is there a way to avoid this error?          
          
```
<template v-for="i in selectedSupplierRef.length - 1" :key="i" >
<q-td :key="'supplier_' + i" :props="props">{{ props.row.supplier_1 }}</q-td>
</template>
```
          
I tried the following, but couldn't retrieve the value of 'i'.

```
<template v-for="i in selectedSupplierRef.length - 1" :key="i" />
<q-td :key="'supplier_' + i" :props="props">{{ props.row.supplier_1 }}</q-td>
```

**Comment by yusufkandemir** [comment, marked as answer]
This is a Vue question.

You can place the v-for in `<q-td>` directly. Since it's the only thing inside the `<template>` tag, this solution makes more sense:
```html
<q-td
  v-for="i in selectedSupplierRef.length - 1"
  :key="'supplier_' + i"
  :props="props"
>
  {{ props.row.supplier_1 }}
</q-td>
```

You can also implement what the error message suggests, and put the key only in the `<q-td>` tag and not the `<template>`:
```html
<template v-for="i in selectedSupplierRef.length - 1">
  <q-td :key="'supplier_' + i" :props="props">{{ props.row.supplier_1 }}</q-td>
</template>
```

**Reply by afurosan** [reply]
Thank you Sir!
